test/sql/test_string_functions/R/test_string_functions

```diff
@@ -236,6 +236,24 @@ select split_part(c1, "##", -3) from t1;
 abc
 -- !result
 
+-- name: test_url_extract_host
+select url_extract_host('https://starrocks.com/test/api/v1');
+-- result:
+starrocks.com
+-- !result
+select url_extract_host('https://starrocks.快速.com/test/api/v1');
+-- result:
+starrocks.快速.com
+-- !result
+select url_extract_host(NULL);
+-- result:
+None
+-- !result
+select url_extract_host('https://starrocks.com/doc?k1=10&k2=3&k1=100');
```

Comment:
What if the argument of url_extract_host is string column, not the const string?

Reply:
> What if the argument of url_extract_host is string column, not the const string?

make sense，i would add some testcase for it @kangkaisen